util/reggen/ip_block.py

```diff
@@ -339,22 +339,20 @@ def get_rnames(self) -> Set[str]:
             ret = ret.union(set(rb.name_to_offset.keys()))
         return ret
 
-
-    def get_signal_by_name_as_dict(self, name) -> Dict:
-        '''Look up and return signal by name'''
-        # This is not super efficient at the moment - but that is OK.
-        for iodir, xput in zip(('inout', 'input', 'output'), self.xputs):
-            for sig in xput:
-                if sig.name == name:
-                    return sig.as_nwt_dict(iodir)
-        else:
-            return None
-
     def get_signals_as_list_of_dicts(self) -> List[Dict]:
         '''Look up and return signal by name'''
-        # This is not super efficient at the moment - but that is OK.
         result = []
         for iodir, xput in zip(('inout', 'input', 'output'), self.xputs):
             for sig in xput:
                 result.append(sig.as_nwt_dict(iodir))
         return result
+
+    def get_signal_by_name_as_dict(self, name: str) -> Optional[Dict]:
```

Comment:
You can now use just Dict instead of Optional[Dict] with the exception you added.

Reply:
ok, done